packages/connect/src/connect.js

```diff
@@ -1,140 +1,193 @@
 import {
-  Component,
-  createContext,
-  createElement,
-  memo,
-  useMemo,
   useState,
   useEffect,
-  useContext,
+  useMemo,
+  memo,
   useRef,
+  createContext,
+  useContext,
 } from "react";
 import { warn } from "@frontity/error";
-import { observe, unobserve, raw, isObservable } from ".";
+import { observe, unobserve } from "@frontity/observer-util";
 
 const COMPONENT = Symbol("owner component");
 
-const context = createContext();
+/**
+ * A React context that stores the store of Frontity Connect.
+ */
+const context = createContext({});
 
+/**
+ * The Provider of the store context, to be used in the root of the React
+ * application.
+ */
 export const Provider = context.Provider;
 
-const hasHooks = typeof useState === "function";
-
+/**
+ * A flag to know when a connected component is being rendered.
+ */
 let isConnected = false;
 
-const mapStateToStores = (state) => {
-  // find store properties and map them to their none observable raw value
-  // to do not trigger none static this.setState calls
-  // from the static getDerivedStateFromProps lifecycle method
-  const component = state[COMPONENT];
-  return Object.keys(component)
-    .map((key) => component[key])
-    .filter(isObservable)
-    .map(raw);
+/**
+ * The default options of connect.
+ */
+const defaultOptions = {
+  /**
+   * Define if `connect` should inject the store props in the component, or only
+   * make it reactive. Useful when the store is accessed via `useConnect` and
+   * the props are passed down an HTML node.
+   */
+  injectProps: true,
 };
 
+/**
+ * Connect a React component with the Frontity connect store and make it
+ * reactive to its changes.
+ *
+ * @param Comp - The React component to be connected.
+ * @param options - The connect options, defined in {@link defaultOptions}.
+ *
+ * @returns The same Component, but reactive and with the store included in the
+ * props.
+ */
 export function connect(Comp, options) {
   const isStatelessComp = !(Comp.prototype && Comp.prototype.isReactComponent);
-
-  let ReactiveComp;
-
-  const defaultOptions = {
-    injectProps: true,
-  };
-
   options = options ? { ...defaultOptions, ...options } : defaultOptions;
 
-  if (isStatelessComp && hasHooks) {
```

Comment:
Why did we remove the `hasHooks` check?

Reply:
Because Frontity ships with React 17, so it always has hooks.

In `react-easy-state` that's not always the case because it supports earlier versions of React.

Reply:
right, makes sense 🙂